### ISSUE TITLE
Unicode ligatures in variable names (suggestion)

### ISSUE THREAD

**Issue by IamMusavaRibica** [satwikkansal/wtfpython#280]
### ▶ Weird unicode ligatures behaviour
Python interpreter interprets unicode ligatures as two or more characters they are made of. If we explicitly declare variables using globals() it doesn't switch. This example uses ﬁ ligature
https://en.wikipedia.org/wiki/Ligature_(writing)#Latin_alphabet

**Output (python 3.9):** 
```py
>>> ﬁg = 4
>>> fig
4
>>> globals()
{'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <class '_frozen_importlib.BuiltinImporter'>, '__spec__': None, '__annotations__': {}, '__builtins__': <module 'builtins' (built-in)>, 'fig': 4}
>>> globals()['ﬁg'] = 6
>>> ﬁg
4
>>> globals()['ﬁg']
6
```
Variable named ﬁg with ligature ﬁ gets saved with both the name of the ligature and "normal" letters. 
We can also use ligatures on previously "normally" defined variables:
```py
>>> fis = 9  # normal f i s
>>> ﬁs  # ligature
9
```
#### 💡 Explanation:
* Unfortunately I cannot explain  as I came up with this on my own. Probably am not the first.
```

**Comment by soft9000** [comment]
Tried it on 3.9.7. Worked as expected:

>>> foo = 6
>>> globals()['foo']
6
>>> globals()['foo'] = 7
>>> foo
7
>>>

**Comment by sebix** [comment]
@soft9000 Not sure if I understand your comment correctly. @IamMusavaRibica point is, that this behaviour you describe does not work if the variable contains ligatures.

I can confirm the behaviour on Python 3.8. Maybe report this as bug at https://bugs.python.org/ instead to check if it's intended behaviour?

**Comment by AchBachir** [comment]
> Tried it on 3.9.7. Worked as expected:
> 
> 
> 
> >>> foo = 6
> 
> >>> globals()['foo']
> 
> 6
> 
> >>> globals()['foo'] = 7
> 
> >>> foo
> 
> 7
> 
> >>> he used ligature **ﬁ** (check this link for more details https://graphemica.com/ﬁ ) instead of normal fi. Basically, they don't have the same characters, so the result must be different.

**Comment by IamMusavaRibica** [comment]
> > Tried it on 3.9.7. Worked as expected:
> > > > > foo = 6
> > 
> > 
> > > > > globals()['foo']
> > 
> > 
> > 6
> > > > > globals()['foo'] = 7
> > 
> > 
> > > > > foo
> > 
> > 
> > 7
> 
> He used ligature **ﬁ** (check this link for more details [https://graphemica.com/ﬁ](https://graphemica.com/%EF%AC%81) ) instead of normal fi. Basically, they don't have the same characters, so the result must be different.

Exactly. My guess is that interpreter takes it as two separate characters. Which is entirely incorrect to me because len() of a ligature is 1. If globals are modified directly then the ligature is retained
